swan-lake/learn-the-language/distinctive-language-features/advanced-general-purpose-language-features.md

```diff
@@ -666,6 +666,102 @@ int cp = ch.toCodePointInt();
 
 In the above code example, **``ch``** is of the ``string:Char`` type. You can also extract the code point of the character by calling the **``toCodePointInt()``** function. This ``string:Char`` subtype is analogous to the XML subtypes like XML elements or text.
 
+## Regular Expressions
+
+Ballerina provides first-class support for regular expressions, represented by the built-in ``RegExp`` type defined in the ``lang.regexp`` module. Additionally, the ``lang.string`` module has a type alias named the same so that one can use either ``regexp:RegExp`` or  ``string:RegExp`` to create variables for regular expressions.
+
+There are two ways to declare regular expressions:
+
+1. Using regular expression template literals:
+
+```ballerina
+string:RegExp reg = re `abc+`;
+```
+In this approach, you can use the ``re`` keyword to define a regular expression pattern using a template literal. This allows for easy expression of regular expressions in a way that is similar to other languages.
```

Comment:
```suggestion
In this approach, you specify the pattern between two backticks, followed by  ``re`` keyword to define a regular expression using a template literal. This allows for easy expression of regular expressions in a way that is similar to other languages.
```
takes a pattern between two slashes, followed by optional flags, after the second slash.